src/main/java/com/marklogic/contentpump/ThreadManager.java

```diff
@@ -217,13 +222,30 @@ public ThreadPoolExecutor initThreadPool() {
         return pool;
     }
 
+    /**
+     * Return the total counts of LocalMapTask that are actively running.
+     * @return
+     */
+    public int getActiveTaskCounts() {
+        int count = 0;
+        for (int i = 0; i < taskList.size(); i++) {
```

Comment:
should we synchronize on taskList or no need to since I'm not sure if this is always called by one thread?